src/hera/workflow.py

```diff
@@ -45,6 +46,8 @@ class Workflow:
         A Dict of labels to attach to the Workflow object metadata
     namespace: Optional[str] = 'default'
         The namespace to use for creating the Workflow.  Defaults to "default"
+    workflow_template_ref: Optional[str] = None
+        The name of the workflowTemplate reference
```

Comment:
What do you think about making the docstring a bit more descriptive? I have a hunch some users may be confused by this field. Some more information around what this field refers to, maybe with a link to Argo Workflows docs, along with mentioning how it will be used over defining a new workflow may help! Thoughts?